pkg/liqo-controller-manager/resource-request-controller/local-resource-monitor.go

```diff
@@ -69,69 +63,45 @@ const (
 	PendingToPending
 )
 
-// SetupBroadcaster initializes all Broadcaster parameters.
-func (b *Broadcaster) SetupBroadcaster(clientset kubernetes.Interface, updater interfaces.UpdaterInterface,
-	resyncPeriod time.Duration, resourceSharingPercentage, offerUpdateThreshold uint64) error {
-	b.allocatable = corev1.ResourceList{}
-	b.resourceSharingPercentage = resourceSharingPercentage
-	b.updateThresholdPercentage = offerUpdateThreshold
-	b.updater = updater
-	b.resourcePodMap = map[string]corev1.ResourceList{}
-	b.lastReadResources = map[string]corev1.ResourceList{}
-	nodesFactory := informers.NewSharedInformerFactoryWithOptions(clientset, resyncPeriod, informers.WithTweakListOptions(virtualNodeFilter))
-	b.nodeInformer = nodesFactory.Core().V1().Nodes().Informer()
-	b.nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    b.onNodeAdd,
-		UpdateFunc: b.onNodeUpdate,
-		DeleteFunc: b.onNodeDelete,
+// NewLocalMonitor creates a new LocalResourceMonitor.
+func NewLocalMonitor(ctx context.Context, clientset kubernetes.Interface, resyncPeriod time.Duration,
+	updater *OfferUpdater) *LocalResourceMonitor {
```

Comment:
Would personally avoid the double link monitor <-> updater. The updater could register itself (a Notifier interface) through an appropriate method exposed by the ResourceReaderInterface.